pkg/controller/node/nodecontroller.go

```diff
@@ -1115,10 +1117,26 @@ func (nc *NodeController) evictPods(node *v1.Node) bool {
 	return nc.zonePodEvictor[utilnode.GetZoneKey(node)].Add(node.Name, string(node.UID))
 }
 
-func (nc *NodeController) markNodeForTainting(node *v1.Node) bool {
+func (nc *NodeController) markNodeForTaintingNotReady(node *v1.Node) bool {
+	nc.evictorLock.Lock()
```

Comment:
You don't need to take a lock if you're not operating on `zone...Tainter`. Same below.